tests/js/session-tests.js

```diff
@@ -929,5 +929,31 @@ module.exports = {
             });
             realm.close();
         });
+    },
+
+    async testAnalyticsSubmission() {
+        const context = node_require('realm/package.json');
+        const analytics = node_require('realm/lib/submit-analytics');
+
+        const payload = await analytics.fetchPlatformData(context, 'TestEvent')
+        .catch((e) => {
+            return Promise.reject("Failed to fetch platform data  [Error: " + e + "]");
+        });
```

Comment:
I would have left this `.catch` out, since you're awaiting the result of the call to `fetchPlatformData` the test will fail if the promise returned gets rejected. The only reason that you would want to add this `catch` would be to "extend" the message, but that probably makes little sense in a test, since you don't expect this to fail at all.

```suggestion
        const payload = await analytics.fetchPlatformData(context, 'TestEvent');
```